src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -61,6 +61,42 @@ Copyright (c) .NET Foundation. All rights reserved.
     <RuntimeIdentifier Condition="'$(PlatformTarget)' == 'x86' or '$(PlatformTarget)' == ''">win7-x86</RuntimeIdentifier>
   </PropertyGroup>
 
+  <!-- Check if SelfContained is global, resets the value to it's original state if it was local.-->
+  <PropertyGroup>
+    <_OriginalSelfContainedValue>$(SelfContained)</_OriginalSelfContainedValue>
+    <SelfContained></SelfContained>
+  </PropertyGroup>
+
+  <Choose>
+    <!-- Assumes a global variable cannot be an undefined variable.-->
+    <When Condition="'$(SelfContained)' != '' and $(SelfContained) == $(_OriginalSelfContainedValue)">
+      <PropertyGroup>
+        <_SelfContainedIsGlobal>true</_SelfContainedIsGlobal>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <_SelfContainedIsGlobal>false</_SelfContainedIsGlobal>
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+
+  <PropertyGroup>
+    <SelfContained>$(_OriginalSelfContainedValue)</SelfContained>
+    <_OriginalSelfContainedValue></_OriginalSelfContainedValue>
+  </PropertyGroup>
+
+  <!-- PubishSelfContained may be set but not change self contained if we are not publishing, or self contained was already set globally. This condition tracks whether it will take effect.-->
+  <PropertyGroup Condition=" '$(_CommandLineDefinedSelfContained)' != 'true' and '$(_SelfContainedIsGlobal)' != 'true' and '$(_IsPublishing)' == 'true' and ('$(PublishSelfContained)' == 'true' or '$(PublishSelfContained)' == 'false')">
```

Comment:
Is the `_CommandLineDefinedSelfContained` redundant here?  I think if `_CommandLineDefinedSelfContained` is true then that would imply `_SelfContainedIsGlobal` would also be true.

Reply:
Ya, thanks